tests/templates/kuttl/tls/check-tls.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+import trino
+import argparse
+import sys
+import yaml
+
+
+if not sys.warnoptions:
+    import warnings
+warnings.simplefilter("ignore")
+
+
+def get_http_connection(host, user):
+    return trino.dbapi.connect(
+        host=host,
+        port=8080,
+        user=user,
+        http_scheme='http',
+    )
+
+
+def get_https_connection(host, user, verify):
+    return trino.dbapi.connect(
+        host=host,
+        port=8443,
+        user=user,
+        http_scheme='https',
+        verify=verify
+    )
+
+
+def get_authenticated_https_connection(host, user, password, verify):
+    return trino.dbapi.connect(
+        host=host,
+        port=8443,
+        user=user,
+        http_scheme='https',
+        auth=trino.auth.BasicAuthentication(user, password),
+        verify=verify
+    )
+
+
+def test_query(conn, query):
+    cursor = conn.cursor()
+    try:
+        cursor.execute(query)
+        result = cursor.fetchone()
+        print("[SUCCESS] Received: " + str(result))
+    except Exception as e:
+        print("[ERROR] " + str(e))
+        exit(-1)
+
+
+def test_query_failure(conn, query, failure_message):
+    cursor = conn.cursor()
+    try:
+        cursor.execute(query)
+        print("[ERROR] " + failure_message)
+        exit(-1)
+    except Exception as e:
```

Comment:
This could be *anything*. Hell, `SyntaxError` is an `Exception` in Python. You probably want to be more precise in what errors you expect.

Reply:
Yeah anything that goes wrong? I mean i only want to fail not take different actions on different exceptions?

Reply:
Yeah, usually you'd want to fail *unless* the error matches the expected error for that case.

Reply:
now catching SSLError (invalid certificates) and HttpError (invalid credentials) explicitly

Reply:
Presumably the specific expected error depends on the test in question. You could pass in the expected error type as an argument to `test_query_failure`.

Reply:
Good point thanks. Done.